### PR TITLE
Feature/31/Line plot Y-axis aggregation functions

### PR DESCRIPTION
**Pull request by d-callan**
think i still need some group fxn tests probably, and to think about how i want to factor the scatter class. im thinking line plot might be its own class, w time series as a subclass.

### CODE REVIEW

File: R/class-plotdata-line.R [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+  }
+  if (attr$xAxisVariable$dataType == 'DATE') {
+    .pd$seriesX <- lapply(.pd$seriesX, format, '%Y-%m-%d')
+  } else { 
+    .pd$seriesX <- lapply(.pd$seriesX, as.character)
+  }
+  if (attr$yAxisVariable$dataType == 'DATE') {
+    .pd$seriesY <- lapply(.pd$seriesY, format, '%Y-%m-%d')
+  } else {
+    .pd$seriesY <- lapply(.pd$seriesY, as.character)
```
Comment (asizemore) [inline R/class-plotdata-line.R:78]: Why do we want to convert numeric to character for the y axis? Is the client converting back to numbers?

**Review by asizemore — APPROVED** [review]
Left one question about the y axis output. Otherwise looks good!

File: R/class-plotdata-line.R (hunk 1 above)
Comment (d-callan) [reply R/class-plotdata-line.R:78]: i did it bc the constraints in the data service claim to be able to handle dates currently. and r can technically get means and medians for dates. idk that to be particularly useful, but i dont know its really wrong either.

Comment (asizemore) [reply R/class-plotdata-line.R:78]: oh i see it now. Cool

Comment (asizemore) [reply R/class-plotdata-line.R:78]: and i see scatter is the same. Makes sense now thank you!

**Pull request merged**